src/main/scala/chisel3/simulator/ChiselSimSettings.scala

```diff
@@ -0,0 +1,174 @@
+package chisel3
+package simulator
+
+import svsim._
+
+/**
+  * Wrapper for unifying compilation and runtime options for different backends
+  *
+  * @param backend                      selected [[svsim.Backend]]
+  * @param verilogPreprocessorDefines   SystemVerilog defines as a map of `MACRO`->definition
+  * @param optimizationStyle
+  * @param availableParallelism
+  * @param defaultTimescale
+  * @param libraryExtensions
+  * @param libraryPaths
+  * @param disabledWarnings
+  * @param disableFatalExitOnWarnings
+  * @param traceStyle
+  * @param outputSplit
+  * @param outputSplitCFuncs
+  * @param enableAllAssertions
+  * @param customSimulationWorkingDirectory
+  * @param verboseCompile
+  * @param verboseRun
+  * @param workspaceRoot
+  * @param workingDirectoryPrefix
+  * @param executionScriptLimit
+  * @param executionScriptEnabled
+  * @param conservativeCommandResolution
+  * @param resetWorkspace                 completely delete workspace folders and their content before running the simulation
+  * @param randomlyInitializeRegisters
+  * @param chiselArgs
+  * @param firtoolArgs
+  */
+case class ChiselSimSettings[B <: Backend](
+  backend:                          B,
+  traceStyle:                       TraceStyle = TraceStyle.NoTrace,
+  verilogPreprocessorDefines:       Seq[(String, Any)] = Seq(),
+  optimizationStyle:                CommonCompilationSettings.OptimizationStyle = CommonCompilationSettings.OptimizationStyle.Default,
+  defaultTimescale:                 Option[CommonCompilationSettings.Timescale] = None,
+  libraryPaths:                     Option[Seq[String]] = None,
+  libraryExtensions:                Option[Seq[String]] = None,
+  disabledWarnings:                 Seq[String] = Seq(/*"WIDTH", */ "STMTDLY"),
+  disableFatalExitOnWarnings:       Boolean = true,
+  outputSplit:                      Option[Int] = Some(20_000),
+  outputSplitCFuncs:                Option[Int] = Some(20_000),
```

Comment:
I think we should avoid making simualtor-specific things part of the generic ChiselSim settings